add: shove

Rafnik02 opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. --> Добавление shove (отталкивания)

Если человека оттолкнуть в стену, то он упадёт и у него начнётся эффект нокдаун на 3 секунды.

Если толкнуть человека в человека, то оба упадут и у них будет эффект нокдаун на 1 секунду.

Если человека толкнуть на стеклянный стол, то стекло стола сломается и будет эффект нокдаун на 4 секунды. Также с обычным столом

Во всех моментах, у человека будут выпадать предметы из рук, но он всё ещё сможет взять в руки что-нибудь после падения и ползти

## Ссылка на предложение/Причина создания ПР
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения --> 
https://discord.com/channels/617003227182792704/755125334097133628/1252011755895525521

Uploading 123231321231.mp4…

Rafnik02 committed:
add: shove

base

Den109G commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -655,9 +666,63 @@
+								"You hear a loud thud.")
+		if(!HAS_TRAIT(target, TRAIT_FLOORED))
+			target.Knockdown(3 SECONDS)
+			addtimer(CALLBACK(target, /mob/living.proc/SetKnockdown, 0), 3 SECONDS) // so you cannot chain stun someone // so you cannot chain stun someone
```
```suggestion
			addtimer(CALLBACK(target, /mob/living.proc/SetKnockdown, 0), 3 SECONDS) // so you cannot chain stun someone
```

Rafnik02 replied:
Сделано

Den109G commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -655,9 +666,63 @@
+	var/moved = target.Move(shove_to, shove_dir)
+	if(!moved) //they got pushed into a dense object
+		add_attack_logs(user, target, "Disarmed into a dense object", ATKLOG_ALL)
+		target.visible_message("<span class='warning'>[user] slams [target] into an obstacle!</span>", \
+								"<span class='userdanger'>You get slammed into the obstacle by [user]!</span>", \
+								"You hear a loud thud.")
```
Лучше переделать на дефайны.
```suggestion
		target.visible_message(span_warning("[user] slams [target]"), \
								span_userdanger("You get slammed into the obstacle by [user]!"), \
								"You hear a loud thud.")
```

Den109G commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -655,9 +666,63 @@
+			target.Slowed(2.5 SECONDS, 1)
+			var/obj/item/I = target.get_active_hand()
+			if(I)
+				to_chat(target, "<span class='warning'>Your grip on [I] loosens!</span>")
```
```suggestion
				to_chat(target, span_warning("Your grip on [I] loosens!"))
```

Rafnik02 committed:
Update _species.dm

Удалил повторяющиеся слова. Поменял некоторые значения на дефайны. Удалил разоружения граба (это заменено шовом)

Rafnik02 committed:
fix error

исправил ошибку, которая выдавала проверка гитхаба

Rafnik02 committed:
Исправил косяки

Упомянул это в предложке, но сам не сделал.

Rafnik02 committed:
fix error

Исправление ошибки, которая нашла проверка гита

Rafnik02 committed:
fix conflict merge

Пробую фиксить конфликт мердж

Rafnik02 committed:
fix conflict merge 2

Пробую фиксить конфликт

Rafnik02 committed:
fix error

Попытка исправить ошибку гита

Rafnik02 committed:
Merge branch 'master220' into shove

Rafnik02 commented:
Сделаю новую ветку.
